ably/state.go

```diff
@@ -532,3 +553,81 @@ func (res *stateResult) Wait() error {
 	}
 	return res.err
 }
+
+// A ConnectionStateV12 identifies the state of an Ably realtime connection.
+type ConnectionStateV12 struct {
+	name string
+}
+
+var (
+	ConnectionStateInitializedV12  = &ConnectionStateV12{name: "INITIALIZED"}
+	ConnectionStateConnectingV12   = &ConnectionStateV12{name: "CONNECTING"}
+	ConnectionStateConnectedV12    = &ConnectionStateV12{name: "CONNECTED"}
+	ConnectionStateDisconnectedV12 = &ConnectionStateV12{name: "DISCONNECTED"}
+	ConnectionStateSuspendedV12    = &ConnectionStateV12{name: "SUSPENDED"}
+	ConnectionStateClosingV12      = &ConnectionStateV12{name: "CLOSING"}
+	ConnectionStateClosedV12       = &ConnectionStateV12{name: "CLOSED"}
+	ConnectionStateFailedV12       = &ConnectionStateV12{name: "FAILED"}
+)
+
+func (e ConnectionStateV12) String() string {
+	return e.name
+}
+
+// A ConnectionEventV12 identifies an event in the lifetime of an Ably realtime
+// connection.
+type ConnectionEventV12 struct {
```

Comment:
This could just be a string:

```go
type ConnectionEvent string

const (
	ConnectionEvent_INITIALIZED ConnectionEvent = "INITIALIZED"
	...
	ConnectionEvent_UPDATED     ConnectionEvent = "UPDATED"
)
```

Reply:
I decided to hide that since it doesn't have string semantics of any kind. It's just an opaque type for token-like, singleton values.

Reply:
👍